### ISSUE TITLE
ARM: `azure ad sp create` does not accept applicationId as a named arg

### ISSUE THREAD

**Issue by colemickens** [Azure/azure-xplat-cli#2217]
```
$ azure ad sp create --help
help:    Create active directory service principal
help:    
help:    Usage: ad sp create [options] <application-id>
help:    
help:    Options:
help:      -h, --help                        output usage information
help:      -v, --verbose                     use verbose output
help:      -vv                               more verbose with debug output
help:      --json                            use json output
help:      --applicationId <application-id>  the application Id for which service principal is created
help:    
help:    Current Mode: arm (Azure Resource Management)
```

then:

```
$ azure ad sp create --applicationId "5b382ef8-a3a3-4e74-87f3-8931d68126b9"

  error: missing required argument `application-id'
```

Also, I feel that the named arg should be `--application-id` to be consistent with other calls, and with the error message. If the named arg is not renamed, then the error message should say `missing required argument 'applicationId'`.

**Comment by yugangw-msft** [comment]
@vivsriaus , please fix it

**Comment by akamalov** [comment]
Confirming a bug...

**Comment by amarzavery** [comment]
Fixed in #2944
